docs/reference/configuration.txt

```diff
@@ -168,7 +168,12 @@ can be configured.
           # on 2.4 and 2.6 is :plain)
           auth_mech: :scram
 
-          # The database or source to authenticate the user against. (default: admin)
+          # When authenticating, look up the user in this database.
+          # This parameter is also called the authentication database or the
+          # auth source.
```

Comment:
How about:

```
Specify the auth source, i.e. the database or other source which contains the user's login credentials.
```

Reply:
My apologies, I approve/request reviews from my tooling which doesn't show comments, so I didn't see this comment prior to approving & requesting review from @comandeo .

The phrase "or other source" wasn't clear to me, therefore I reviewed the definition of auth source. In https://docs.mongodb.com/ruby-driver/master/reference/authentication/#auth-source (which matched my existing understanding) we say:

> A user’s auth source is the database where that user’s authentication credentials are stored.

This claims that auth source is always the database.

However, I then checked server LDAP auth docs at https://docs.mongodb.com/manual/core/security-ldap-external/#user-management, which say:

> With LDAP authorization, user creation and management occurs on the LDAP server. MongoDB requires creation of roles on the admin database, with the name of each role exactly matching a LDAP group Distinguished Name (DN). This is in contrast to MongoDB managed authorization, which requires creating users on the $external database.

Meaning, for LDAP authentication, users aren't in fact stored in any MongoDB database, and our driver documentation in this regard isn't quite correct.

I replaced my text with your suggestion and I added a note to refer to server documentation for figuring out what auth sources are allowed, what do you think?

Reply:
I've made a refinement to your addition which includes a link to the server docs page. Please check it.

Reply:
:+1: Thank you for that.